packages/swapper/src/swappers/thorchain/utils/makeSwapMemo.ts

```diff
@@ -0,0 +1,43 @@
+import { adapters } from '@shapeshiftoss/caip'
+
+import { SwapError, SwapErrorTypes } from '../../../api'
+
+// BTC (and likely other utxo coins) can only support up to 80 character memos
+const MAX_LENGTH = 80
```

Comment:
We are going to need to think about how to do a reverse fuzzy lookup on abbreviated assets in order to turn it into a caip assetId if there is not a ticket for that already

Reply:
oh thats a fun one

Reply:
@elmutt i agree with @kaladinlight  and that logic absolutely needs to go into the caip adapter